tests/federation/test_federation_client.py

```diff
@@ -83,7 +83,7 @@ def test_get_room_state(self):
         )
 
         # mock up the response, and have the agent return it
-        self._mock_agent.request.return_value = defer.succeed(
+        self._mock_agent.request.side_effect = lambda *args, **kwargs: defer.succeed(
```

Comment:
Curious why we stick with `defer.succeed` for this one?

Reply:
Good question. I think the rationale was that `_mock_agent` is supposed be an `IAgent` and twisted defines `IAgent.request` to return a `Deferred[IResponse]`

Reply:
Ah, yes that makes sense! 👍 This truly *does* return a `Deferred`!